nx/lib/nx.ex

```diff
@@ -10694,7 +10799,19 @@ defmodule Nx do
     {output_shape, output_names} = Nx.Shape.dot(s1, c1, names1, b1, s2, c2, names2, b2)
 
     out = %{t1 | type: output_type, names: output_names, shape: output_shape}
-    impl!(t1, t2).dot(out, t1, c1, b1, t2, c2, b2)
+
+    t1 = devectorize(t1)
+    t2 = devectorize(t2)
+    out = devectorize(out)
+
+    c1 = Enum.map(c1, &(&1 + offset))
+    c2 = Enum.map(c2, &(&1 + offset))
+    b1 = offset_axes ++ Enum.map(b1, &(&1 + offset))
+    b2 = offset_axes ++ Enum.map(b2, &(&1 + offset))
+
+    res = impl!(t1, t2).dot(out, t1, c1, b1, t2, c2, b2)
```

Comment:
Let's also wrap this on `if offset > 0` to avoid additional work unless necessary.